### DISCUSSION TITLE
Get download url from JSON file

### DISCUSSION THREAD

**Discussion by Birdie0** [ScoopInstaller/Scoop#5653] [Help and Questions] [answered]
Can I extract download url from JSON file with JSONPath in the same way I can do that with hash or version?

Example JSON: `https://example.com/check.json`

```json
{
  "version": "1.2.3",
  "md5": "4738b33da0417b39857b6856e3f9a49b",
  "url": "https://example.com/files/{random-letters}.zip"
}
```

- `{random-letters}` represent random unique alphanumeric string

App manifest file:

```json
{
    "version": "1.2.3",
    "description": "test",
    "homepage": "https://example.com",
    "license": "Freeware",
    "url": "https://example.com/files/aaaa.zip",
    "hash": "md5:4738b33da0417b39857b6856e3f9a49b",
    "bin": "app.exe",
    "shortcuts": [
        [
            "app.exe",
            "App"
        ]
    ],
    "checkver": {
        "url": "https://example.com/check.json",
        "jsonpath": "$.version"
    },
    "autoupdate": {
        // "url": "https://example.com/check.json",
        // "jsonpath": "$.url",
        "hash": {
            "url": "https://example.com/check.json",
            "jsonpath": "$.md5"
        }
    }
}
```

**Comment by rashil2000** [comment, marked as answer]
No, the autoupdate URL is a string that contains substitution variables - it does not use checkver's extraction logic. 

However, you can use the checkver field itself, to fetch both the URL and the version. See https://github.com/ScoopInstaller/Main/blob/master/bucket/gcc.json for example.

**Reply by Birdie0** [reply]
I see, it seems only option I have now is to parse JSON with regex.
